app/controllers/uploads_controller.rb

```diff
@@ -9,48 +9,61 @@ def index
   end
 
   def new
-    csv_type = params[:csv_type]
-
-    @upload = Upload.new(csv_type: csv_type)
-    @upload.skip_lines = defaults(csv_type)['skip_lines']
-
+    @upload = new_upload(params[:csv_type])
     return if @upload.csv_type.blank? || @upload.csv_type_check?
```

Comment:
In light of your changes (removing the generic upload - I should have taken this out.